src/modules/navigator/gpsfailure.cpp

```diff
@@ -83,9 +83,24 @@ GpsFailure::on_active()
 	case GPSF_STATE_LOITER: {
 			/* Position controller does not run in this mode:
 			 * navigator has to publish an attitude setpoint */
+
+			struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
+			struct vehicle_status_s *stat = _navigator->get_vstatus();
+
 			vehicle_attitude_setpoint_s att_sp = {};
 			att_sp.timestamp = hrt_absolute_time();
-			att_sp.roll_body = math::radians(_param_nav_gpsf_r.get());
+
+			//This determines if the fixedwing vehicle is landing or not, and if so it forces the wings level
+			//Otherwise the plane goes to the parameter set fixed bank loiter setting
+
+			if (stat->nav_state == vehicle_status_s::NAVIGATION_STATE_AUTO_LANDGPSFAIL
```

Comment:
Checking the nav_state is redundant. https://github.com/PX4/Firmware/blob/89e1f478ac9f3863df4f5613cf111222ae9c7749/src/modules/navigator/navigator_main.cpp#L617